docs/src/usage/authentication.md

```diff
@@ -0,0 +1,86 @@
+# Authentication
+
+```
+rover-auth 0.0.0
+🔐  Manage authentication
+
+USAGE:
+    rover auth <SUBCOMMAND>
+
+FLAGS:
+    -h, --help       Prints help information
+    -V, --version    Prints version information
+
+SUBCOMMANDS:
+    api-key    🔑 Use an Account or Graph API Key
+    help      Prints this message or the help of the given subcommand(s)
+```
+
+For many of the CLI operations, you will not need to authenticate at all!
+
+However, to use Apollo Studio services, such as checks, either in local
+development or in CI, you'll want to grant the CLI permission to perform a set
+of operations on your graphs. You can do this via API tokens.
+
+Currently, there are 2 levels of permissions one can generate:
+
+| Type | Generatable by | Perms | Recommended Use |
+|---------|---------------------|-------------------------------|---------------------------|
+| Account | Apollo Studio User | Same as the account | Local development machine |
+| Graph | Apollo Studio Admin | Full perms for a single graph | CI |
+
+You can authenticate the CLI via a command or an environment variable, using
+either type of token. 
+
+**Environment variables take precedence over the `auth` command. This means 
+that if you set an enviroment variable, it will override any configuration
+a previously run `auth` command set.**
+
+## Commands
+
+```
+rover auth api-key
+[INFO] Go to https://engine.apollographql.com/user-settings and create a new Personal API Key.
```

Comment:
We don't want to use `engine` anywhere in the new CLI. This should be `studio.apollographql.com` anywhere that is `engine`

Reply:
got it- i pulled this from the `rust` repo, but am happy to rename :)